src/lib/net/SecureListenSocket.h

```diff
@@ -16,8 +18,14 @@ class IDataSocket;
 class SecureListenSocket : public TCPListenSocket
 {
 public:
-  SecureListenSocket(IEventQueue *events, SocketMultiplexer *socketMultiplexer, IArchNetwork::EAddressFamily family);
+  SecureListenSocket(
+      IEventQueue *events, SocketMultiplexer *socketMultiplexer, IArchNetwork::EAddressFamily family,
+      SecurityLevel securityLevel = SecurityLevel::PlainText
+  );
 
   // IListenSocket overrides
   virtual IDataSocket *accept();
+
+private:
+  SecurityLevel m_securityLevel;
```

Comment:
As this is only set in ctor, const makes sense rather than it being mutable.
```suggestion
  const SecurityLevel m_securityLevel;
```